src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -82,8 +84,10 @@ public static class CreateMutationBuilder
             // we find that the input object has already been created for the entity.
             inputs.Add(input.Name, input);
 
-            // Generate fields for related entities only if multiple mutations are supported for the database flavor.
-            if (DoesRelationalDBSupportMultipleCreate(databaseType))
+            // Generate fields for related entities when
+            // 1. Multiple mutation operations are supported for the database type.
+            // 2. Multiple mutation operations are enabled.
+            if (isMultipleCreateOperationEnabled && DoesRelationalDBSupportMultipleCreate(databaseType))
```

Comment:
Since, https://github.com/Azure/data-api-builder/pull/2116#discussion_r1536631102 and this comment are about the same thing, we can have furhter discussion on that thread. 

Resolving this one.